core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -949,7 +949,16 @@ OuterGroupBy operation has to be applied only in the following cases
         }
         builder
       } else {
-        val query = getDimOnlyQuery(bestDimCandidates, requestModel, queryGenVersion)
+        val isTotalRowQueryWithOneReturnedRow =
+          requestModel.isDimDriven && requestModel.includeRowCount && requestModel.reportingRequest.rowsPerPage == 1 && factBestCandidateOption.isEmpty
+
+        var totalRowRequestModel = requestModel
+
+        if(isTotalRowQueryWithOneReturnedRow)
+          totalRowRequestModel = requestModel.copy(dimSortByMap = Map.empty, factSortByMap = Map.empty, hasDimSortBy = false, hasFactSortBy = false)
```

Comment:
If all you are doing is clearing out the sorting for dim and fact, you should do this in the curator.  Not here.  If you want to change how the query is executed by the pipeline, e.g. single vs multi-engine then you should do it here.